stage #0/module #3/lesson #3/content.md

```diff
@@ -0,0 +1,246 @@
+    (Index from 0 to str.length()-1)
+    The chars in a string are identified by "index" numbers. In "Hello" the leftmost char (H) is at index 0, the next char (e) is at index 1, and so on.
+    
+     2.1. int indexOf(char ch );
+     
+       This method returns the index within this string of the first occurrence of the specified character or -1, if the character does not occur:
+     
+        String mjc = new String("Welcome to MJC!");
+         
+        System.out.print("Found 't' first at position : ");
+         
+        // Initial index of 't' will print
+        // prints 8
+        System.out.println(mjc.indexOf('t')); 
+        
+     2.2.  int indexOf(char ch, int strt);
+        
+       This method returns the index within this string of the first occurrence of the specified character, starting the search at the specified index or -1, if the character does not occur.
+        
+        String mjc = new String("Welcome to MJC school!");
+                
+        System.out.print("Found 'o' after 7 index at position : ");
+                
+        // prints 9
+        System.out.println(mjc.indexOf('o', 7));   
+        
+  3. String Substring
+  
+     - str.substring(start)
+     - Chars beginning at index start
+     - Through the end of the string
+   
+      The substring() method picks out a part of string using index numbers to identify the desired part. The simplest form, substring(int start) takes a start index number and returns a new string made of the chars starting at that index and running through the end of the string
+   
+         String a = str.substring(1);  // a is "arry Potter" (i.e. starting at index 1)
+         String b = str.substring(2);  // b is "rry Potter"
+         String c = str.substring(3);  // c is "ry Potter"
+       The start (begin index) parameter must be within the range of source string, otherwise you would see the following exception:
+    
+         java.lang.StringIndexOutOfBoundsException: String index out of range:
+   
+ 
+
+**In all cases of calling methods that modify a string, a new object of type String is created!**
+###Why String is Really Special?
+ 1. String + Concatenation.
+ 
+    The + (plus) operator between strings puts them together to make a new, bigger string. The bigger string is just the chars of the first string put together with the chars of the second string.
+      
+        String bestSeason = "Sum" + "mer";
+    Keep in mind that Strings not made of the letters a-z. For example chars can be punctuation:
+     
+        String currentDate = "The 31st" + " " + "of January" + ", ";
+ 2. String is immutable. 
+    Since string literals with the same contents share storage in the common pool, Java's String is designed to be immutable. That is, once a String is constructed, its contents cannot be modified. Otherwise, the other String references sharing the same storage location will be affected by the change, which can be unpredictable and therefore is undesirable.
```

Comment:
seems that either this part could go later after the string pool is explained or to move string pool before this section, or maybe even combine it with the creation of a string

Reply:
Yes, String Pool will be in another stage. Have removed this description